src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/TerminateDebugHandler.ts

```diff
@@ -51,11 +62,15 @@ export async function onDidTerminateDebugSession(
   logger.logVerbose(`[DEBUGGER] Terminating debugging session with PID ${targetPid}...`);
 
   try {
-    process.kill(targetPid);
+    if (processExists(targetPid)) {
+      process.kill(targetPid);
```

Comment:
If the previous way we killed the process "fails" why would the extra processExists checks succeed?

Reply:
> If the previous way we killed the process "fails"

Are you referring to the on in `processExists`? If so, that one is a no-op.

Reply:
I guess I'm just asking why have the processExist check at all?